test/e2e/framework/gateways.go

```diff
@@ -198,3 +198,32 @@ func (f *Framework) GatewayCleanup() {
 		}
 	}
 }
+
+// Perform a gateway failover.
+// The failover for the real environment will crash the gateway node.
+// THe failover for the kind environment will set the "false" label on the gw node.
+func (f *Framework) DoFailover(cluster ClusterIndex, gwNode, gwPod string) {
+	provider := DetectProvider(cluster, gwNode)
+
+	if provider == "kind" {
+		f.SetGatewayLabelOnNode(cluster, gwNode, false)
+	} else {
+		cmd := []string{"sh", "-c", "echo 1 > /proc/sys/kernel/sysrq && echo b > /proc/sysrq-trigger"}
+
+		_, _, err := f.ExecWithOptions(&ExecOptions{
+			Command:       cmd,
+			Namespace:     TestContext.SubmarinerNamespace,
+			PodName:       gwPod,
+			ContainerName: SubmarinerGateway,
+			CaptureStdout: false,
+			CaptureStderr: true,
+		}, cluster)
+		if err != nil {
+			if strings.HasPrefix(err.Error(), "unable to upgrade connection: container not found") {
```

Comment:
Relying on a specific error message is a bit fragile. I'm wondering if we can check a specific  error type using the K8s error API functions, eg `apierrors.IsNotFound(err)`. Can you print out the full error with `%#v` formatting so we can see its type and other info?

Reply:
Will check that.

Reply:
@tpantelis 
This is the output of the error type:
`&errors.errorString{s:"unable to upgrade connection: container not found (\"submariner-gateway\")"}`

Could you guide me on how to move on, please?

Reply:
OK so it's not a specific error type we can key off of. So we'll have to check for a specific string as you're doing. However the underlying [code](https://github.com/kubernetes/apimachinery/blob/f8159af4957e1f55a9ba27d0ce48ad7a5fa41d02/pkg/util/httpstream/spdy/roundtripper.go#L350) constructs a generalized error string prefixed with "_unable to upgrade connection_"  so I think the underlying error message is most relevant so I suggest checking `strings.Contains(err.Error(), "container not found")`.

Reply:
Done